packages/waku/src/router/base-types.ts

```diff
@@ -0,0 +1,22 @@
+import type { createPages } from './create-pages.js';
+import type { CollectPaths } from './inferred-router-types.js';
+
+type AnyRouter = ReturnType<typeof createPages>;
+
+type RouterSpec<Router extends AnyRouter> = Awaited<
+  NonNullable<Router['DO_NOT_USE_ready']>
+>[number];
+
+// eslint-disable-next-line @typescript-eslint/no-empty-object-type
+export interface RouteConfig {
+  // routes to be overridden by users
+}
```

Comment:
This is fine in this file. Maybe only this.

Reply:
This and the re-export of PathsForRouter are now the only things in this file